libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -0,0 +1,301 @@
+- (void)removeDelegate:(id<SFSDKWindowManagerDelegate>)delegate
+{
+    @synchronized (self) {
+        [_delegates removeObject:[NSValue valueWithNonretainedObject:delegate]];
+    }
+}
+
+- (void)enumerateDelegates:(void (^)(id<SFSDKWindowManagerDelegate> delegate))block
+{
+    @synchronized(self) {
+        [_delegates.allObjects enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+            id<SFSDKWindowManagerDelegate> delegate = [obj nonretainedObjectValue];
+            if (delegate) {
+                if (block) block(delegate);
+            }
+        }];
+    }
+}
+#pragma mark - SFSDKWindowContainerDelegate
+- (void)windowEnable:(SFSDKWindowContainer *_Nonnull)window animated:(BOOL)animated withCompletion:(void (^)(void))completion{
+
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:willEnableWindow:)]){
+            [delegate windowManager:self willEnableWindow:window];
+        }
+    }];
+    
+    if (animated) {
+        __weak typeof (self) weakSelf = self;
+        UIViewPropertyAnimator *animator = [[UIViewPropertyAnimator alloc] initWithDuration:0.25 curve:UIViewAnimationCurveEaseInOut animations:^{
+            [weakSelf makeOpagueWithCompletion:window completion:completion];
+        }];
+        [animator startAnimation];
+    } else {
+        [self makeOpagueWithCompletion:window completion:completion];
+    }
+}
+
+- (void)windowDisable:(SFSDKWindowContainer *)window animated:(BOOL)animated withCompletion:(void (^)(void))completion{
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:willDisableWindow:)]){
+            [delegate windowManager:self willDisableWindow:window];
+        }
+    }];
+   
+    if (animated) {
+        __weak typeof (self) weakSelf = self;
+        UIViewPropertyAnimator *animator = [[UIViewPropertyAnimator alloc] initWithDuration:0.25 curve:UIViewAnimationCurveEaseInOut animations:^{
+            [weakSelf makeTransparentWithCompletion:window completion:completion];
+        }];
+        
+        [animator startAnimation];
+    } else {
+        [self makeTransparentWithCompletion:window completion:completion];
+    }
+    
+}
+
+
+#pragma mark - private methods
+- (void)makeTransparentWithCompletion:(SFSDKWindowContainer *)window completion:(void (^)(void))completion {
+    window.window.alpha = 0.0; //make Transparent
+    
+    [self updateKeyWindow];
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:didDisableWindow:)]){
+            [delegate windowManager:self didDisableWindow:window];
+        }
+    }];
+    if (completion)
+        completion();
+}
+
+- (void)makeOpagueWithCompletion:(SFSDKWindowContainer *)window completion:(void (^)(void))completion {
+    window.window.alpha = 1.0; //make Opague
```

Comment:
So you make one window opaque and then loop through all window looking for a non-transparent window and make the first one found the key window. So if the app created other (non-transparent) windows, couldn't one of those be made the key window instead of the one you explicitly enabled?